### PR TITLE
Remove 8.11 node version inference code

### PR DESCRIPTION
**Pull request by thecoop**
Remove the mapping from version to transport version needed to communicate with pre-8.11 nodes. This is needed to remove Version from DiscoveryNode entirely (see #115434)

### CODE REVIEW

**Comment by thecoop** [comment]
@edsavage am I correct in thinking that 8.18+ nodes will always have ML and Transform config version attributes on them?

**Comment by thecoop** [comment]
@elasticmachine update branch

**Comment by thecoop** [comment]
@elasticmachine update branch

**Comment by edsavage** [comment]
> @edsavage am I correct in thinking that 8.18+ nodes will always have ML and Transform config version attributes on them?

Yes, that's correct.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/TransformConfigVersion.java [hunk 1]
```diff
@@ -331,10 +331,8 @@ public static Tuple<TransformConfigVersion, TransformConfigVersion> getMinMaxTra
 
     public static TransformConfigVersion getTransformConfigVersionForNode(DiscoveryNode node) {
         String transformConfigVerStr = node.getAttributes().get(TRANSFORM_CONFIG_VERSION_NODE_ATTR);
-        if (transformConfigVerStr != null) {
-            return fromString(transformConfigVerStr);
-        }
-        return fromId(node.getPre811VersionId().orElseThrow(() -> new IllegalStateException("getting legacy version id not possible")));
```
Comment (ldematte) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/TransformConfigVersion.java:337]: Nit: this is the last usage of getPre811VersionId, can we remove it? Or do you want to do that in another PR?

**Review by ldematte — APPROVED** [review]
LGTM.
One question: this PR assumes that v9 nodes will never talk to "old" v8 nodes, but just with v8.last.
My understanding is that this is what we want to do, but do we have a barrier in place? Is it enforced somewhere e.g. in JoinRequest? 
(it's just curiosity on my side)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/TransformConfigVersion.java (hunk 1 above)
Comment (thecoop) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/TransformConfigVersion.java:337]: I'm removing it in https://github.com/elastic/elasticsearch/pull/115434

**Comment by thecoop** [comment]
Yes, v9 will only be able to talk to 8.18. This is enforced by `TransportVersions.MINIMUM_COMPATIBLE`, which we won't be able to set until we know what the transport version for 8.18 is.

**Comment by thecoop** [comment]
@elasticmachine rerun elasticsearch-ci/part-2
